lib/fastlane/plugin/wpmreleasetoolkit/helper/metadata_download_helper.rb

```diff
@@ -100,6 +95,33 @@ def delete_existing_metadata(target_locale)
       def get_target_file_path(locale, file_name)
         "#{@target_folder}/#{locale}/#{file_name}"
       end
+
+      private
+
+      def handle_glotpress_download(response:, locale:, is_source:)
+        case response.code
+        when '200'
+          # All good, parse the result
+          UI.success("Successfully downloaded `#{locale}`.")
+          @alternates.clear
+          loc_data = JSON.parse(response.body) rescue loc_data = nil
+          parse_data(locale, loc_data, is_source)
+          reparse_alternates(target_locale, loc_data, is_source) unless @alternates.length == 0
+        when '301'
+          # Follow the redirect
+          UI.message("Received 301 for `#{locale}`. Following redirect...")
+          download(locale, response.header['location'], is_source)
+        when '429'
+          # We got rate-limited, offer to try again
+          if UI.confirm("Retry downloading `#{locale}` after receiving 429 from the API?")
+            download(locale, response.uri, is_source)
+          else
+            UI.error("Abandoning `#{locale}` download as requested.")
+          end
+        else
+          UI.error("Received unexpected #{response.code} from request to URI #{response.uri}.")
```

Comment:
Should we stop the execution when encounter this error?

Reply:
Good question 🤔 I'm unsure. Is an error on one locale worth blocking the entire process? From one point of view it might be, so that we can go on and fix it. From another, HTTP comes with occasional transport errors and it would be good to update _some_ metadata at least.

Reply:
Middle-ground: ask the user 😄 

```suggestion
          message = "Received unexpected #{response.code} from request to URI #{response.uri}."
          UI.abort_with_message!(message) unless UI.confirm("#{message} Continue anyway?")
```

Reply:
Nice, so there's less chances it'll be missed, too.

Reply:
Just a note to say that I "tested" this before merging:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/1218433/188756964-4dec2e10-c5a6-41c7-a71d-3682eb986e97.png">